include/libdnf/base/transaction.hpp

```diff
@@ -81,6 +81,12 @@ class Transaction {
     /// @return the transaction groups.
     std::vector<libdnf::base::TransactionGroup> & get_transaction_groups() const;
 
+    /// Check the transaction by running it with RPMTRANS_FLAG_TEST. This check
+    /// is performed automatically by run(); it is redundant to call test()
+    /// before calling run().
```

Comment:
There is one thing that deserves to be documented - import of GPG keys. If I am correct - test in DNF5 and in DNF also imports GPG keys. It is let say incorrect - test is not supposed to do permanent changes. On the other hand - we need that behavior for workflow in system upgrade and offline upgrades.

Reply:
Thanks, it's a good idea to make that explicit. Is the language correct here?

    /// Check the transaction by running it with RPMTRANS_FLAG_TEST. Packages
    /// will be downloaded, public keys will be installed, and transaction
    /// checks will be performed, but no changes to the package set will be
    /// made. These checks are performed automatically by run(); it is
    /// redundant to call test() before calling run().

Reply:
I believe that this is incorrect - `Packages will be downloaded`. The method does not download packages.

Reply:
About `public keys will be installed` - this is not wrong, but it performs complex tasks related to signed rpm keys verification. First of all - the test of keys is only applied when configuration requires it. The workflow - it tries to verify signature using already imported keys in rpm-db. If it fails then it downloads keys, request approval for import (callback), than it imports keys and request the verification of the RPM signature.
`install` vs. `import` keys into RPM DB - I somehow do not prefer `install`.

Reply:
`to the package set` - wat about to add `installed in the statement - `to the installed package set`.

Reply:
> I believe that this is incorrect - Packages will be downloaded. The method does not download packages.

Thanks, yes of course, oops!

> About public keys will be installed - this is not wrong...

Changed to "the import of any necessary public keys will be requested".

 > to the package set - what about to add installed in the statement - to the installed package set`.

Done.